### PR TITLE
Whack a mole game

### PR DESCRIPTION
**Pull request by angy113**
testspassed
progress

### CODE REVIEW

File: src/main/java/org/minions/devfund/angela/Main.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        final WhackAMole whackAMole = new WhackAMole(numAttempts, gridDimension);
+        Random randomI = new Random();
+        Random randomJ = new Random();
+        final int molesQuantity = 10;
+        for (int i = 0; i < molesQuantity; i++) {
+            if (!whackAMole.place(randomI.nextInt(gridDimension), randomJ.nextInt(gridDimension))) {
+                i--;
+            }
+        }
+        for (int i = 0; i < numAttempts; i++) {
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/angela/Main.java]: MINOR Reduce the total number of break and continue statements in this loop to use at most one. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS135)

File: src/main/java/org/minions/devfund/angela/WhackAMole.java [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+            for (int j = 0; j < moleGrid.length; j++) {
+                if (moleGrid[i][j] == 'M') {
+                    builder.append("* ");
+                } else {
+                    builder.append(moleGrid[i][j]).append(" ");
+                }
+            }
+            builder.append("\n");
+        }
+        System.out.println(builder);
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/angela/WhackAMole.java]: this line of code is no longer necessary.

File: src/main/java/org/minions/devfund/angela/WhackAMole.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+     */
+    public String printGrid() {
+        final StringBuilder builder = new StringBuilder();
+        for (int i = 0; i < moleGrid.length; i++) {
+            for (int j = 0; j < moleGrid.length; j++) {
+                builder.append(moleGrid[i][j]).append(" ");
+            }
+            builder.append("\n");
+        }
+        System.out.println(builder);
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/angela/WhackAMole.java]: this line of code is no longer necessary.

File: src/test/java/org/minions/devfund/angela/WhackAMoleTest.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
+package org.minions.devfund.angela;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+
+import static junit.framework.TestCase.assertTrue;
```
Comment (carledriss) [inline src/test/java/org/minions/devfund/angela/WhackAMoleTest.java]: That package will be deprecated in the feature
You should use 
```java
import static org.junit.Assert.assertTrue;
```

File: src/main/java/org/minions/devfund/angela/WhackAMole.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+    public WhackAMole(int numAttempts, int gridDimension) {
+        molesLeft = 0;
+        score = 0;
+        attemptsLeft = numAttempts;
+        moleGrid = new char[gridDimension][gridDimension];
+        for (int i = 0; i < gridDimension; i++) {
+            for (int j = 0; j < gridDimension; j++) {
+                moleGrid[i][j] = '*';
+            }
+        }
```
Comment (carledriss) [inline src/main/java/org/minions/devfund/angela/WhackAMole.java]: you can use the following code.
```java
for (char[] row : moleGrid) {
    Arrays.fill(row, '*');
}
```

Comment (angy113) [reply src/main/java/org/minions/devfund/angela/WhackAMole.java]: Done

File: src/test/java/org/minions/devfund/angela/WhackAMoleTest.java (hunk 4 above)
Comment (angy113) [reply src/test/java/org/minions/devfund/angela/WhackAMoleTest.java]: Done

File: src/main/java/org/minions/devfund/angela/WhackAMole.java (hunk 3 above)
Comment (angy113) [reply src/main/java/org/minions/devfund/angela/WhackAMole.java]: Done

File: src/main/java/org/minions/devfund/angela/WhackAMole.java (hunk 2 above)
Comment (angy113) [reply src/main/java/org/minions/devfund/angela/WhackAMole.java]: Done

**Comment by carledriss** [comment]
LGTM! 👍

**Pull request merged**